### PR TITLE
Set remaining model lower limits to zero and make sure demo loop runs with integral approach off

### LINKED ISSUE #356: Fix discrete model selection and model applicability
When sampling a discrete interaction, we sample a process using the per-process cross section calculated at the pre-step energy and then use the post-step energy to select the applicable model. Because of inconsistencies between the energy grid endpoints of the cross section tables and the model limits, it was possible for a process to be sampled but not have a model that applied at that energy. To get around this and make sure a model is always selected for the sampled process (but is this actually necessary, or can we just return an invalid model ID if none are applicable?), all our models currently have their lower applicable energy set to zero. This is wrong, but should produce correct results as long as the integral approach is used. However, without rejecting interactions based on the cross section at the post-step energy, we can end up e.g. with a stopped positron selecting bremsstrahlung because of the incorrect model energy bounds. We should figure out how to handle these inconsistencies and correctly set the model energy limits.

### PR DESCRIPTION
**Pull request by amandalund**
Closes #356.

### CODE REVIEW

File: src/celeritas/phys/ParticleTrackView.hh [hunk 1]
```diff
@@ -340,7 +340,7 @@ CELER_FUNCTION units::MevMomentumSq ParticleTrackView::momentum_sq() const
 {
     const real_type energy = this->energy().value();
     real_type result = energy * energy + 2 * this->mass().value() * energy;
-    CELER_ENSURE(result > 0);
+    CELER_ENSURE(result >= 0);
```
Comment (amandalund) [inline src/celeritas/phys/ParticleTrackView.hh:343]: With integral approach off this failed when a positron sampled ionization before coming to rest and momentum was calculated in the Moller-Bhabha interactor constructor.

Comment (sethrj) [reply src/celeritas/phys/ParticleTrackView.hh:343]: That should still be positive for a positron right? The only thing with zero momentum should be a photon "at rest"...

Comment (amandalund) [reply src/celeritas/phys/ParticleTrackView.hh:343]: Hmm if the kinetic energy is zero then the momentum should be zero, right?

Comment (sethrj) [reply src/celeritas/phys/ParticleTrackView.hh:343]: Oh goodness, I was thinking total energy. You're quite right, ignore me!

**Review by sethrj — APPROVED** [review]
Thanks @amandalund .

**Pull request merged**